Source/AwsCommonRuntimeKit/crt/CRTError.swift

```diff
@@ -1,18 +1,30 @@
-//  Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
-//  SPDX-License-Identifier: Apache-2.0.
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0.
 import AwsCCommon
 
-public struct CRTError: Error {
+extension CRTError {
+    public init(fromErrorCode errorCode: Int32) {
+        self = CRTError(rawValue: errorCode) ?? .UNKNOWN_ERROR_CODE
+    }
+
+    public init(fromErrorCode errorCode: Int) {
+        self = CRTError(rawValue: Int32(errorCode)) ?? .UNKNOWN_ERROR_CODE
+    }
 
-    public let errorCode: Int32
+    /// Use aws_last_error() as the error code
+    public init() {
+        self.init(fromErrorCode: aws_last_error())
+    }
 
-    public let errorMessage: String
+    public var errorCode: Int32 {
```

Comment:
trivial: I'd be fine if this was just `Int` instead of `Int32`

Reply:
Int size can be different depending on the arch, but I think it is okay to use Int here.

Reply:
Swift expects it to be Int32 instead of Int for passing to C functions. I tried using an Int and then created a simple init function which takes an int and converts it to Int32 to use it with C calls.